src/amuse/datamodel/particle_attributes.py

```diff
@@ -691,6 +691,20 @@ def LagrangianRadii(stars, unit_converter=None, mf=[0.01,0.02,0.05,0.1,0.2,0.5,0
     0.856966667972 length
     """
     import bisect
+    if unit_converter is "auto":
+        # Try to determine the right unit base, using the mass
+        mass_base_unit = stars.mass.unit.base[0][1]
+        if mass_base_unit is units.kg:
+            # Set converter lengths to be something that seems reasonable
+            converter_mass = stars.total_mass() / len(stars)
```

Comment:
I think maybe just the total mass would be better

Reply:
It probably doesn't matter in any real use case.
My thinking here was that this will make the individual particle mass as close as possible to 1.0.